packages/keystatic/src/component-blocks/cloud-image-preview.tsx

```diff
@@ -2,159 +2,410 @@
+                    display: 'flex',
+                    alignItems: 'center',
+                    justifyContent: 'center',
+                  })}
+                >
+                  <ProgressCircle
+                    size="small"
+                    aria-label="Checking…"
+                    isIndeterminate
+                  />
+                </div>
+              ) : state.src ? (
+                <ActionButton
+                  prominence="low"
+                  onPress={() => setState(cleanImageData())}
+                >
+                  <Icon src={xIcon} />
+                </ActionButton>
+              ) : null
+            }
+          />
+          {status === 'good' ? (
+            <>
+              <TextArea
+                label="Alt Text"
+                value={state.alt}
+                onChange={alt => setState(state => ({ ...state, alt }))}
+              />
+              <Flex gap="regular" alignItems="end">
+                <TextField
+                  label="Width"
+                  value={state.width}
+                  width="scale.1000"
+                  onChange={width => setState(state => ({ ...state, width }))}
+                />
+                <TextField
+                  label="Height"
+                  value={state.height}
+                  width="scale.1000"
+                  onChange={height => setState(state => ({ ...state, height }))}
+                />
+                {dimensions &&
+                (`${dimensions.width}` !== state.width ||
+                  `${dimensions.height}` !== state.height) ? (
+                  <div>
+                    <ActionButton
+                      isDisabled={dimensions === null}
+                      alignSelf="start"
+                      onPress={() => {
+                        setState(state => ({
+                          ...state,
+                          height: dimensions.height,
+                          width: dimensions.width,
+                        }));
+                      }}
+                    >
+                      Set to{' '}
+                      {dimensions
+                        ? `${dimensions.width} x ${dimensions.height}`
+                        : null}
+                    </ActionButton>
+                  </div>
+                ) : null}
+              </Flex>
+            </>
+          ) : null}
+        </Flex>
+      </Content>
+      <ButtonGroup>
+        <Button onPress={onClose}>Cancel</Button>
+        <Button
+          prominence="high"
+          type="submit"
+          form="example-form"
+          isDisabled={status !== 'good'}
+        >
+          {image ? 'Done' : 'Insert'}
+        </Button>
+      </ButtonGroup>
+    </Dialog>
+  );
+}
+
+function Placeholder({
+  onChange,
+  onRemove,
+}: {
+  onChange: (data: ImageData) => void;
+  onRemove: () => void;
+}) {
+  return (
+    <NotEditable>
+      <Flex gap="regular">
+        <Box flex="1">
```

Comment:
I think this is unnecessary, you should be able to move the flex prop to the `Button` e.g.

```jsx
<Flex>
  <DialogTrigger>
    <Button flex />
  </DialogTrigger>
</Flex>
```